src/ion/liveranges.rs

```diff
@@ -476,7 +476,8 @@ impl<'a, F: Function> Env<'a, F> {
                 // register can be used multiple times in the same
                 // instruction is with an early-use and a late-def. Anything
                 // else is a user error.
-                let mut operand_rewrites: FxHashMap<usize, Operand> = FxHashMap::default();
+                let mut operand_rewrites = core::mem::take(&mut self.ctx.scratch_operand_rewrites);
```

Comment:
Do we need to do a `core::mem::take` on each instruction, and put it back below? Perhaps this could be hoisted out of the loop over instructions?